src/app/utils/route-utils.ts

```diff
@@ -31,3 +33,8 @@ export abstract class RouteUtils {
     return RouteUtils.ENABLED_PARA_TIMES
   }
 }
+
+export const addressParamLoader = async ({ params }: LoaderFunctionArgs) => {
+  const address = await getOasisAddress(params.address!)
+  return address
```

Comment:
Perhaps it would be easier to understand if this returned `{oasisAddress, ethAddress?}`